src/primitives/transaction.h

```diff
@@ -15,6 +15,11 @@ static const int SERIALIZE_TRANSACTION_NO_WITNESS = 0x40000000;
 
 static const int WITNESS_SCALE_FACTOR = 4;
 
+// Opt-in chain rejection based on bits 29 and 30 of the nVersion field (for replay protection)
+static const int TX_VERSION_MASK = 0x1fffffff;
+static const int TX_VERSION_REJECT_SEGWIT2X = 0x20000000;
```

Comment:
If this PR is accepted, I hope it will also be merged by Bitcoin Unlimited and Bitcoin Classic. Could we name this `TX_VERSION_REJECT_2X` instead of `TX_VERSION_REJECT_SEGWIT2X`? This would clarify that it's about the 2x hard-fork, not specially about the Segwit2x agreement. Bitcoin Unlimited for example is blind to the Segwit2x agreement, but will follow the 2x chain if it has most proof of work.